runtime/databricks/automl_runtime/sklearn/compose/target.py

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright (C) 2022 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from sklearn.base import ClassifierMixin, BaseEstimator, clone
+from sklearn.utils.validation import check_is_fitted
+from sklearn.preprocessing import FunctionTransformer
+
+
+class TransformedTargetClassifier(ClassifierMixin, BaseEstimator):
+    """Meta-estimator to classify on a transformed target.
+
+    Useful when the underlying classifier only accepts specified format of targets.
+    For example, xgboost requires the target values to be in {0, 1, ..., num_class -1}
+    since 1.6.0.
+
+    The implementation is similar to sklearn.compose.TransformedTargetRegressor.
+
+    Parameters
+    ----------
+    classifier : object
+        Classifier object such as derived from sklearn.base.ClassifierMixin.
+        This classifier will automatically be cloned each time prior to fitting.
+    transformer : object, default=None
+        Estimator object that is designed to transform targets/labels, such as
+        transformers defined in sklearn.preprocessing._label.
+        If transformer is None, the transformer will be an identity
+        transformer. Note that the transformer will be cloned during fitting.
+        Notice that the transformer should work for the specific `y` to be
+        passed in `fit`, otherwise the `fit` call can fail.
+    """
+
+    def __init__(self, classifier, *, transformer=None):
+        self.classifier = classifier
+        self.transformer = transformer
+
+    def _fit_transformer(self, y):
+        if self.transformer is None:
+            self.transformer = FunctionTransformer()  # identity transformer
+        self.transformer_ = clone(self.transformer)
```

Comment:
Why do we need to `clone` the transformer?

Reply:
Actually I'm just following what sklearn's `TransformedTargetRegressor` does. My guess is that it might be generally a convention to avoid modifying the passed instance.